yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,203 @@
+	METRIC_FREE_DISK_SPACE           = "free_disk_space"
+	METRIC_TOTAL_DISK_SPACE          = "total_disk_space"
+	FREE_DISK_SPACE_THREASHOLD       = 10
+	REPLICATION_GUARDRAIL_ALERT_MSG  = "Replication is enabled on the cluster, enable it only once the import-data is done ingesting data to use the disk only for the data import operation and keep the cluster stable."
+)
+
+type TargetDBForMonitorHealth interface {
+	GetYBServers() (bool, []*tgtdb.TargetConf, error)
+	RemoveConnectionsForHosts([]string) error
+	NumOfLogicalReplicationSlots() (int64, error)
+	GetClusterMetrics() (map[string]tgtdb.NodeMetrics, error) // node_uuid:metric_name:metric_value
+}
+
+// bool variable is for indicating if a node is down or up
+var nodeStatuses map[string]bool
+
+func MonitorTargetYBHealth(yb TargetDBForMonitorHealth, skipDiskUsageHealthChecks utils.BoolStr, skipReplicationChecks utils.BoolStr, skipNodeHealthChecks utils.BoolStr, displayAlertFunc func(alertMsg string)) error {
+	nodeStatuses = make(map[string]bool)
+
+	var err error
+	loadBalancerEnabled, servers, err := yb.GetYBServers()
+	if err != nil {
+		return fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	for _, server := range servers {
+		nodeStatuses[server.Host] = true
+	}
+
+	var nodeAlert string
+	for {
+		nodeAlert, err = monitorNodesStatusAndAdapt(yb, loadBalancerEnabled, skipNodeHealthChecks)
+		if err != nil {
+			log.Errorf("error monitoring the node status and adapt: %v", err)
+		}
+
+		//skipping the disk usage monitoring for now, util the metrics function changes are made
+		err = monitorDiskUsageAndAbort(yb, true)
+		if err != nil {
+			log.Errorf("error monitoring the disk and memory: %v", err)
+		}
+
+		err = monitorReplicationOnTarget(yb, skipReplicationChecks)
+		if err != nil {
+			log.Errorf("error monitoring the replication: %v", err)
+		}
+
+		if nodeAlert != "" {
+			alertMsg := color.RedString(fmt.Sprintf("Alert!\n%s\n", nodeAlert))
+			displayAlertFunc(alertMsg)
+		}
+		time.Sleep(time.Duration(MONITOR_HEALTH_FREQUENCY_SECONDS) * time.Second)
+	}
+}
+
+/*
+1. if node goes down
+remove its connections from pool and if any current importBatch is running it will fail and automatic retry will take care of using the rest conns
+
+2. if node come back
+the new connections will automatically go to that node -- handled via conn-pool
+
+3. load balancer on the cluster
+No node status checks will happen as we put the load on the load balancer IP and it will take care of this adaptation automatically
+*/
+func monitorNodesStatusAndAdapt(yb TargetDBForMonitorHealth, loadBalancerEnabled bool, skip utils.BoolStr) (string, error) {
+	if bool(skip) || loadBalancerEnabled {
+		return "", nil
+	}
+	_, currentNodes, err := yb.GetYBServers()
+	if err != nil {
+		return "", fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	downNodes := make([]string, 0)
+
+	for node, up := range nodeStatuses {
+		if up && !lo.ContainsBy(currentNodes, func(t *tgtdb.TargetConf) bool {
```

Comment:
if nodeStatus was down, but now it has come back up (i.e. it is present in currentNodes), probably worth indicating to the user via a message on console that the node is back up?